osc-feedback.js

```diff
@@ -0,0 +1,74 @@
+const osc = require('osc');
+
+let buffer = Buffer.alloc(0);
+
+function getCompleteMessageLength(buffer) {
+    try {
+        const packet = osc.readPacket(buffer, {});
+        return osc.writePacket(packet).length;
+    } catch (err) {
+        // Handle incomplete message
+        return buffer.length + 1; // Ensure the message length exceeds buffer length to wait for more data
+    }
+}
+
+async function parseOscMessages(root, buffer) {
+    const packets = [];
+
+    while (buffer.length > 0) {
+        const messageLength = getCompleteMessageLength(buffer);
+        if (messageLength <= buffer.length) {
+            const message = buffer.slice(0, messageLength);
+            buffer = buffer.slice(messageLength);
+
+            try {
+                let packet = osc.readPacket(message, {});
+                packets.push(packet);
+
+            } catch (err) {
+                root.log('error', `Error parsing OSC message: ${err.message}. Data: ${message}`);
+            }
+        } else {
+            break; // Wait for more data
+        }
+    }
+
+    return { remainingBuffer: buffer, packets };
+}
+
+async function onDataHandler(root, data) {
+    try {
+        buffer = Buffer.concat([buffer, data]);
```

Comment:
I'm not a fan of this `buffer` being a global variable. There isn't any technically wrong, but it is a bit of a code smell. 
Its not a big problem if it stays though.

One thing that probably should be done, is that when the connection is opened, this buffer should be emptied, as if there is anything left in the buffer that will cause 'corruption' in the stream

Reply:
The reasoning behind the buffer is because some messages can come through in mulitple packets (especially blobs), doing this allows the full messages to be recieved and reconstructed. However in this implication this might indeed be redundant so I will re-scope it.